xds/src/main/java/io/grpc/xds/sds/trust/SdsTrustManagerFactory.java

```diff
@@ -47,19 +46,17 @@
   private static final Logger logger = Logger.getLogger(SdsTrustManagerFactory.class.getName());
   private SdsX509TrustManager sdsX509TrustManager;
 
-  /** Constructor that loads certs from a file. */
-  public SdsTrustManagerFactory(
-      String certsFile, @Nullable CertificateValidationContext certContext)
+  /** Constructor constructs from a {@link CertificateValidationContext}. */
+  public SdsTrustManagerFactory(CertificateValidationContext certContext)
       throws CertificateException, IOException, CertStoreException {
-    this(CertificateUtils.toX509Certificates(new File(certsFile)), certContext);
+    checkNotNull(certContext, "certContext");
+    String certsFile = getTrustedCaFromCertContext(certContext);
+    createSdsX509TrustManager(
+        CertificateUtils.toX509Certificates(new File(certsFile)), certContext);
   }
 
-  /** Constructor that takes array of certs. */
-  public SdsTrustManagerFactory(
-      X509Certificate[] certs, @Nullable CertificateValidationContext certContext)
-      throws CertStoreException {
-    checkNotNull(certs, "certs");
-    createSdsX509TrustManager(certs, certContext);
+  private static String getTrustedCaFromCertContext(CertificateValidationContext certContext) {
+    return certContext != null ? certContext.getTrustedCa().getFilename() : null;
```

Comment:
we don't handle other 2 types? (inline bytes or string)

Reply:
No. The checks are already there in `SslContextSecretVolumeSecretProvider.validate*` [methods](https://github.com/grpc/grpc-java/blob/master/xds/src/main/java/io/grpc/xds/sds/SslContextSecretVolumeSecretProvider.java#L66) and this is not yet used by the control plane AFAIK

Reply:
this method is called from internal public api, you can't assume that it will be only called from `SslContextSecretVolumeSecretProvider`. also why it checks null if it cannot be null? it needs to have `@Nullable`. 
those 2 cases are completely opposite approach (one is not handling possible case with the class, the other is handling impossible case) which made me very confused.

Reply:
Fixed the logic to make it consistent and clear for this class alone. Also fixed format/style